etc/testing/circle/workloads/go-test-results/egress/main.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgconn"
+	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/jmoiron/sqlx"
+	gotestresults "github.com/pachyderm/pachyderm/v2/etc/testing/circle/workloads/go-test-results"
+	"github.com/pkg/errors"
+)
+
+var logger = log.Default()
+var postgresqlUser = os.Getenv("POSTGRESQL_USER")
+var postgresqlPassword = os.Getenv("POSTGRESQL_PASSWORD")
+var postgresqlHost = os.Getenv("POSTGRESQL_HOST")
+
+const (
+	jobInfoFileName string = "JobInfo.json"
+)
+
+type TestResultLine struct {
+	Time     time.Time `json:"Time"`
+	Action   string    `json:"Action"`
+	TestName string    `json:"Test"`
+	Package  string    `json:"Package"`
+	Elapsed  float32   `json:"Elapsed"`
+	Output   string    `json:"Output"`
+}
+
+// This is built and runs in a pachyderm pipeline to egress data uploaded to pachyderm to postgresql.
+func main() {
+	logger.Println("Beginning egress transform of data to sql DB")
+	inputFolder := os.Args[1]
+
+	db, err := sqlx.Open("pgx", fmt.Sprintf("user=%s password=%s host=%s port=5432 database=ci_metrics", postgresqlUser, postgresqlPassword, postgresqlHost))
+	if err != nil {
+		logger.Fatalf("Failed opening db connection %v", err)
+	}
+	// filpathWalkDir does not evaluate the PFS symlink
+	sym, err := filepath.EvalSymlinks(inputFolder)
+	if err != nil {
+		logger.Fatal(err)
+	}
+	jobInfoPaths := make(map[string]gotestresults.JobInfo)
+	testResultPaths := []string{}
+	err = filepath.WalkDir(sym, func(path string, d fs.DirEntry, err error) error {
+		if !d.IsDir() {
```

Comment:
This isn't strictly problematic, but I'd invert the logic here; `if d.IsDir() { return nil }` and then un-indent the bulk of the code.

I think you need to check the passed-in error before doing anything else.   (`if err != nil { return err }` is likely adequate.)

Reply:
good call